modin/backends/pandas/query_compiler.py

```diff
@@ -1009,6 +1009,81 @@ def rolling_aggregate(self, rolling_args, func, *args, **kwargs):
         )
         return self.__constructor__(new_modin_frame)
 
+    def unstack(self, level, fill_value):
+        if not isinstance(self.index, pandas.MultiIndex) or (
+            isinstance(self.index, pandas.MultiIndex)
+            and is_list_like(level)
+            and len(level) == self.index.nlevels
+        ):
+            axis = 1
+            new_columns = ["__reduced__"]
+            need_reindex = True
+        else:
+            axis = 0
+            new_columns = None
+            need_reindex = False
+
+        def map_func(df):
+            return pandas.DataFrame(df.unstack(level=level, fill_value=fill_value))
+
+        is_all_multi_list = False
+        if (
+            isinstance(self.index, pandas.MultiIndex)
+            and isinstance(self.columns, pandas.MultiIndex)
+            and is_list_like(level)
+            and len(level) == self.index.nlevels
+            and need_reindex
+        ):
+            is_all_multi_list = True
+            real_cols_bkp = self.columns
+            self.columns = np.arange(len(self.columns))
+
+        new_modin_frame = self._modin_frame._apply_full_axis(
+            axis, map_func, new_columns=new_columns
+        )
+        result = self.__constructor__(new_modin_frame)
+
+        if is_all_multi_list:
+            result = result.sort_index()
+            index_ordered = result.index
+            real_cols = list(real_cols_bkp)
+
+            real_cols_extended = [
+                col for col in real_cols for _ in np.arange(len(self.index))
+            ]
+            reverted_index = [
+                col + tup[1:] for col, tup in zip(real_cols_extended, index_ordered)
+            ]
+            result.index = pandas.MultiIndex.from_tuples(reverted_index)
+            self.columns = real_cols_bkp
+            return result
+
+        if need_reindex:
+            if isinstance(self.index, pandas.MultiIndex):
+                index_tuples = self.index.tolist()
+                columns_list = self.columns.tolist()
```

Comment:
same question about conversion to `list`

Reply:
fixed